src/daft-catalog/src/python.rs

```diff
@@ -3,100 +3,160 @@
+            Ok(())
         })
     }
 
-    fn list_tables(&self, pattern: Option<String>) -> Result<Vec<Identifier>> {
+    fn has_namespace(&self, ident: &Identifier) -> CatalogResult<bool> {
+        Python::with_gil(|py| {
+            let catalog = self.0.bind(py);
+            let ident = PyIdentifier(ident.clone()).to_pyobj(py)?;
+            Ok(catalog
+                .call_method1(intern!(py, "_has_namespace"), (ident,))?
+                .extract()?)
+        })
+    }
+
+    fn drop_namespace(&mut self, ident: &Identifier) -> CatalogResult<()> {
+        Python::with_gil(|py| {
+            let catalog = self.0.bind(py);
+            let ident = PyIdentifier(ident.clone()).to_pyobj(py)?;
+            catalog.call_method1(intern!(py, "_drop_namespace"), (ident,))?;
+            Ok(())
+        })
+    }
+
+    fn list_namespaces(&self, prefix: Option<&Identifier>) -> CatalogResult<Vec<Identifier>> {
         Python::with_gil(|py| {
-            // catalog = 'python catalog object'
             let catalog = self.0.bind(py);
-            // call the list_tables method with the pattern
-            let tables = catalog.getattr("list_tables")?.call1((pattern,))?;
-            // convert the Python list of strings to Vec<Identifier>
-            let tables: Vec<String> = tables.extract()?;
-            let identifiers = tables
+            let prefix = prefix
+                .map(|p| PyIdentifier(p.clone()).to_pyobj(py))
+                .transpose()?;
+            let namespaces_py = catalog.call_method1(intern!(py, "_list_namespaces"), (prefix,))?;
+            Ok(namespaces_py
+                .downcast::<PyList>()
+                .expect("Catalog._list_namespaces must return a list")
                 .into_iter()
-                .map(|table| Identifier::from_sql(&table, false))
-                .collect::<std::result::Result<Vec<Identifier>, _>>()?;
-            Ok(identifiers)
+                .map(|ident| {
+                    Ok(ident
+                        .getattr(intern!(py, "_ident"))?
+                        .extract::<PyIdentifier>()?
+                        .0)
+                })
+                .collect::<PyResult<Vec<_>>>()?)
         })
     }
 
-    fn to_py(&self, py: Python<'_>) -> PyResult<PyObject> {
-        self.0.extract(py)
+    fn create_table(&mut self, ident: &Identifier, schema: &SchemaRef) -> CatalogResult<TableRef> {
+        Python::with_gil(|py| {
+            let catalog = self.0.bind(py);
+            let ident = PyIdentifier(ident.clone()).to_pyobj(py)?;
+            let schema = PySchema {
+                schema: schema.clone(),
+            };
+            let schema_py = py
+                .import(intern!(py, "daft.schema"))?
+                .getattr(intern!(py, "Schema"))?
+                .call_method1(intern!(py, "_from_pyschema"), (schema,))?;
+
+            let table = catalog.call_method1(intern!(py, "_create_table"), (ident, schema_py))?;
```

Comment:
I see why you wanted function overloads based upon how this file is written, but why not just call to `create_table` here? I don't believe all these `_create` etc. methods are necessary.

Reply:
I suppose ultimately it doesn't matter, since the public methods take in a superset of the types of the abstract ones. But since we will only be passing in the types accepted by the abstract methods, I figured it would make sense to go directly to them.